lib/ClangImporter/ClangImporter.cpp

```diff
@@ -859,6 +840,59 @@ bool ClangImporter::addSearchPath(StringRef newSearchPath, bool isFramework,
   return false;
 }
 
+namespace {
+class ClangImporterDependencyCollector : public clang::DependencyCollector
+{
+  const ClangImporterOptions &Options;
+  const bool BridgingHeaderIsPCH;
+  public:
+  ClangImporterDependencyCollector(const ClangImporterOptions &Options)
+    : Options(Options),
+      BridgingHeaderIsPCH(
+          llvm::sys::path::extension(
+              Options.BridgingHeader).endswith(PCH_EXTENSION))
+  {}
+
+  bool isClangImporterSpecialName(StringRef Filename) {
+    using ImporterImpl = ClangImporter::Implementation;
+    return (Filename == ImporterImpl::moduleImportBufferName
+            || Filename == ImporterImpl::bridgingHeaderBufferName
+            // Currently ignoring dependency on bridging .pch files because
+            // they are temporaries; if and when they are no longer
+            // temporaries, this condition should be removed.
+            || (BridgingHeaderIsPCH
+                && Filename == Options.BridgingHeader));
```

Comment:
I was hoping we could avoid it completely. Maybe a more generic answer is to allow passing an array of filenames to exclude? That would leak the knowledge about pch vs. non-pch outside of the importer though.